stomp_moveit/src/stomp_smoothing_adapter.cpp

```diff
@@ -104,23 +107,23 @@ class StompSmoothingAdapter : public planning_request_adapter::PlanningRequestAd
     }
 
     // Initialize STOMP Planner
-    stomp_moveit::StompPlanner stompPlanner(req.group_name, group_config_it->second, ps->getRobotModel());
-    if(!stompPlanner.canServiceRequest(seed_req))
+    stomp_moveit::StompPlanner stomp_planner(req.group_name, group_config_it->second, ps->getRobotModel());
+    if(!stomp_planner.canServiceRequest(seed_req))
     {
       ROS_ERROR("STOMP planner unable to service request");
       res.error_code_.val = moveit_msgs::MoveItErrorCodes::FAILURE;
       return false;
     }
 
     // Setup Planning Context
-    stompPlanner.clear();
-    stompPlanner.setPlanningScene(ps);
-    stompPlanner.setMotionPlanRequest(seed_req);
+    stomp_planner.clear();
+    stomp_planner.setPlanningScene(ps);
+    stomp_planner.setMotionPlanRequest(seed_req);
 
     // Solve
     ROS_DEBUG("Smoothing result trajectory with STOMP");
     planning_interface::MotionPlanDetailedResponse stomp_res;
-    bool success = stompPlanner.solve(stomp_res);
+    bool success = stomp_planner.solve(stomp_res);
     if (success)
     {
       // Successful responses always contain one entry for trajectory_ and proccessing_time_
```

Comment:
In some recent testing that we did, when stomp did not succeed it would return false deeming the entire plan invalid even when ompl had found a valid trajectory.  This behavior felt rather inconvenient and so I'm thinking that the default behavior here should be to return a smoothed trajectory when stomp does succeed but return the  unchanged trajectory passed to the smoother adapter when stomp fails and maybe print a warning.  In both cases the stomp adapter would return true.

Reply:
Can you reproduce one of those cases? I'm curious if the plan returned by OMPL is actually collision-free? I have seen some cases where this is not true because simplification in OMPL creates a path that actually collides with an edge but due to finite discretization is found to be collision-free.

If indeed the trajectory is completely valid, we might have to rethink the way STOMP is seeded as it does not normally regress from a valid solution to an invalid one (I remember fixing that at some point in the past). Maybe it just needs a zero-step where it validates the seed and sets `collision_free_` (or whatever the state variable) to `true`?

Reply:
Silently passing on the input when STOMP fails doesn't sound like a desirable property to me for a planning adapter (or any algorithm).

A warning is easily overlooked, especially in a system which is not monitored closely by humans.